framework/source/class/qx/core/Id.js

```diff
@@ -0,0 +1,166 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2017 Zenesis Limited, http://www.zenesis.com
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * John Spackman (john.spackman@zenesis.com, @johnspackman)
+
+************************************************************************ */
+
+/**
+ * Provides a registry of top level objects
+ */
+qx.Class.define("qx.core.Id", {
+  extend: qx.core.Object,
+  include: [ qx.core.MObjectId ],
+  type: "singleton",
+  
+  members: {
+    
+    __registeredObjects: null,
+    __registeredIdHashes: null,
+
+    /*
+     * @Override
+     */
+    _createObjectImpl: function(id) {
+      if (this.__registeredObjects) {
+        var obj = this.__registeredObjects[id];
+        if (obj !== undefined) {
+          return obj;
+        }
+      }
+      
+      switch(id) {
+        case "application":
+          return qx.core.Init.getApplication() || undefined;
+      }
+      
+      return; // undefined
```

Comment:
For the benefit of editors that flag as an error `return` with a value mixed with `return` without a value in a single function, please `return undefined;` here instead.